pygridgen/grid.py

```diff
@@ -972,6 +985,18 @@ def generate_grid(self):
 
         super(Gridgen, self).__init__(x, y)
 
+    def to_spec(self):
+        focus = Focus.from_spec(self.focus.to_spec())
+        output_dict = {'xbry': self.xbry, 'ybry': self.ybry,
```

Comment:
There are a lot more attributes that define the Gridgen object (e.g., ul_idx, proj, nnodes, precision, etc. Have a look at `Gridgen.__init__` in grid.py. We need to cache all of the user-input that eventually determine where the nodes are placed.